keps/sig-cli/2227-kubectl-default-container/README.md

```diff
@@ -0,0 +1,279 @@
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Introduce an optional annotation for consumption by certain kubectl commands which will allow picking a default container.
+
+## Motivation
+
+Pods are composed of one or more containers. This leads to an extra effort by the operators when they need to run a command on a Pod that needs to target a specific container.
+
+It gets worse because aside from a warning, this couples the default exec container to the container order.
+
+As an example, with Service Mesh now a Pod can always have two containers: the main application and the sidecar. The container startup order impacts on which is the "default" container nowadays, leading to command executions against wrong containers.
+
+Having a well-known annotation that specifies what's the default container of that Pod reduces operation efforts and improves the user experience.
+
+> However, it gets worse because aside from a warning, this couples the default exec container to the container order. 
+> The container ordering also happens to have an impact on container startup order. We have started offering an option 
+> to inject our sidecar as the first container (previously, it was the last one), which has resulted in users running 
+> kubectl exec and getting the "wrong" container.
+
+Quoted from [kubernetes #96986](https://github.com/kubernetes/kubernetes/issues/96986) opened by @howardjohn 
+
+### Goals
+
+- Provide a way for consumers (CLI, operators) to know which is the default Container of a Pod
+- Deprecate the already in use annotation `kubectl.kubernetes.io/default-logs-container`
+
+### Non-Goals
+
+- If the cli is not kubectl, we don't determine which is the default container.
+- Automatically define/create the default container annotation. This is an user operation.
+
+## Proposal
+
+Define a default well-known annotation for Pods called `kubectl.kubernetes.io/default-container` which points to Kubectl (and other commands) what's the default container to be used when the command needs this information.
+
+Also this KEP proposes the change of behavior of kubectl commands that relies on the --container flag to read and make use of this annotation when the flag is not provided.
+
+We don't have a general default container name attribute for pod, and this would change pod spec and is not acceptable. Why not use `kubectl.kubernetes.io/default-container` as the default container name annotation for all four commands above?
```

Comment:
Maybe this can be moved to motivation, removing the last phrase ("Why not use") as you're proposing this on the first statement of ## Proposal, or just removing the whole block seems good also :)

Reply:
removed.